backtracking/m_coloringproblem.py

```diff
@@ -0,0 +1,89 @@
+# Python3 program for the above approach
+
+# Number of vertices in the graph
+# define 4 4
+
+# check if the colored
+# graph is safe or not
+
+
+def isSafe(graph, color):
+
+    # check for every edge
+    for i in range(4):
+        for j in range(i + 1, 4):
+            if graph[i][j] and color[j] == color[i]:
+                return False
+    return True
```

Comment:
On your machine run `python3 -m doctest -v backtracking/m_coloringproblem.py` to see the results of these doctests.

```suggestion
test_graph_0 = [
        [0, 1, 1, 1],
        [1, 0, 1, 0],
        [1, 1, 0, 1],
        [1, 0, 1, 0],
    ]

test_graph_1 = [
        [1, 1, 1, 1],
        [1, 0, 1, 0],
        [1, 1, 0, 1],
        [1, 0, 1, 0],
    ]

def is_safe(graph: list[list[int]], color: list[int]) -> bool:
    """
    Check for every edge to see if it is pink.  FIX ME!!!

    >>> is_safe([[]], [])
    True
    >>> is_safe(test_graph_0, [])
    Traceback (most recent call last):
      ...
    IndexError: list index out of range
    >>> is_safe(test_graph_0, list(range(3)))
    Traceback (most recent call last):
      ...
    IndexError: list index out of range
    >>> is_safe(test_graph_0, list(range(4)))
    True
    >>> is_safe(test_graph_1, [])
    Traceback (most recent call last):
      ...
    IndexError: list index out of range
    >>> is_safe(test_graph_1, list(range(3)))
    Traceback (most recent call last):
      ...
    IndexError: list index out of range
    >>> is_safe(test_graph_1, list(range(4)))
    True
    """
    for i in range(len(graph)):
        for j in range(i + 1, len(graph[0])):
            if graph[i][j] == 1 and color[i] == color[j]:
                return False
    return True
```

Reply:
We should have at least one test that returns `False`.